smpplib/client.py

```diff
@@ -176,6 +179,10 @@ def send_pdu(self, p):
                 consts.DESCRIPTIONS[consts.SMPP_ESME_RINVBNDSTS],
             ))
 
+        if self.max_outstanding_operations and p.command[-5:] != '_resp':
+            while self.outstanding_operations >= self.max_outstanding_operations:
+                sleep(.1)
```

Comment:
I'd probably use `threading.Event.wait()` to block here. Then response handling code could flag the event to unblock the loop here. This way I wouldn't need the `sleep`

Reply:
Never used events before, will take a look a that, thank you.